core/src/main/scala/matryoshka/Nu.scala

```diff
@@ -20,25 +20,28 @@ import Recursive.ops._
 
 import scalaz._, Scalaz._
 
-final case class Nu[F[_]](unNu: Id ~> F, a: Id[_])
+/** This is for coinductive (potentially infinite) recursive structures, models
+  * the concept of “codata”, aka, the “greatest fixed point”.
+  */
+sealed abstract class Nu[F[_]]
+private final case class Nu0[F[_], A](unNu: A => F[A], a: A) extends Nu[F]
```

Comment:
My eyes totally glossed over the `object Nu …` part before and I read `new Nu[F] …` as being the definition of `ana`. But yeah, with that `apply`, totally just as good readability-wise. Switching.